src/openforms/appointments/contrib/qmatic/plugin.py

```diff
@@ -0,0 +1,175 @@
+import json
+import logging
+from datetime import date, datetime, time
+
+from openforms.appointments.contrib.qmatic.models import QmaticConfig
+
+from ..base import (
+    AppointmentClient,
+    AppointmentLocation,
+    AppointmentProduct,
+    BasePlugin,
+)
+
+logger = logging.getLogger(__name__)
+
+
+class Plugin(BasePlugin):
+    """
+    Plugin for Qmatic Orchestra Calendar Public Appointment API (july 2017)
+    """
+
+    def __init__(self):
+        config = QmaticConfig.get_solo()
+        if not config.service:
+            logger.warning("No service defined for Qmatic")
+            return
+
+        self.client = config.service.build_client()
+        self.api_root = config.service.api_root
+
+    def get_available_products(self, current_products: list = None) -> list:
+        """
+        NOTE: The API does not support making an appointment for multiple
+        products. The `current_products` argument is ignored.
+        """
+        try:
+            response = self.client.get(f"{self.api_root}services")
+            if response.status_code != 200:
+                raise Exception(response.text)
+        except Exception as e:
+            logger.exception(e)
+            return []
+
+        return [
+            AppointmentProduct(entry["publicId"], entry["name"])
+            for entry in response.json()["serviceList"]
+        ]
+
+    def get_locations(self, products: list) -> list:
+        if len(products) != 1:
+            return []
+
+        product_id = products[0].identifier
+
+        try:
+            response = self.client.get(f"{self.api_root}services/{product_id}/branches")
+            if response.status_code != 200:
+                raise Exception(response.text)
```

Comment:
Should rather be `response.raise_for_status()` which fails for anything that's HTTP 400+